patroni/postgresql/__init__.py

```diff
@@ -853,7 +853,28 @@ def configure_server_parameters(self):
     def move_data_directory(self):
         if os.path.isdir(self._data_dir) and not self.is_running():
             try:
-                new_name = '{0}_{1}'.format(self._data_dir, time.strftime('%Y-%m-%d-%H-%M-%S'))
+                postfix = time.strftime('%Y-%m-%d-%H-%M-%S')
+                # let's see if pg_xlog|pg_wal is a symlink, in this case we
+                # should move the target
+                for pg_wal_dir in ('pg_xlog', 'pg_wal'):
+                    pg_wal_path = os.path.join(self._data_dir, pg_wal_dir)
+                    if os.path.exists(pg_wal_path) and os.path.islink(pg_wal_path):
+                        pg_wal_realpath = os.path.realpath(pg_wal_path)
+                        logger.info('renaming WAL directory: %s', pg_wal_realpath)
+                        new_name = '{0}_{1}'.format(pg_wal_realpath, postfix)
+                        os.rename(pg_wal_realpath, new_name)
```

Comment:
I think it would be great not only to rename, but update the source of symlink.
The same applies to tablespaces.